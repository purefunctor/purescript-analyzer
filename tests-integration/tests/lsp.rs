// Do not edit! See build.rs

#[rustfmt::skip]
#[test]
fn test_001_definition_local_main() {
    let mut compiler = tests_integration::load_compiler(std::path::Path::new("fixtures/lsp/001_definition_local"));
    let Some(id) = compiler.runtime.module_file("Main") else {
        return;
    };
    let report = tests_integration::lsp::report(&mut compiler, id);
    insta::assert_snapshot!(report);
}

#[rustfmt::skip]
#[test]
fn test_002_definition_import_main() {
    let mut compiler = tests_integration::load_compiler(std::path::Path::new("fixtures/lsp/002_definition_import"));
    let Some(id) = compiler.runtime.module_file("Main") else {
        return;
    };
    let report = tests_integration::lsp::report(&mut compiler, id);
    insta::assert_snapshot!(report);
}

#[rustfmt::skip]
#[test]
fn test_003_hover_local_main() {
    let mut compiler = tests_integration::load_compiler(std::path::Path::new("fixtures/lsp/003_hover_local"));
    let Some(id) = compiler.runtime.module_file("Main") else {
        return;
    };
    let report = tests_integration::lsp::report(&mut compiler, id);
    insta::assert_snapshot!(report);
}

#[rustfmt::skip]
#[test]
fn test_004_hover_import_main() {
    let mut compiler = tests_integration::load_compiler(std::path::Path::new("fixtures/lsp/004_hover_import"));
    let Some(id) = compiler.runtime.module_file("Main") else {
        return;
    };
    let report = tests_integration::lsp::report(&mut compiler, id);
    insta::assert_snapshot!(report);
}
