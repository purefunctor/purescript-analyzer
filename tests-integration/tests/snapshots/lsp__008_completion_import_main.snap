---
source: tests-integration/tests/lsp.rs
expression: report
---
Completion at Position { line: 7, character: 7 }

```
ref = l
--     ^
```

╭───────────────────┬─────────────────┬─────────────────────┬───────────────────┬───────────────────┬───────────────────┬─────────────────────────╮
│  label            │  label_detail   │  label_description  │  sort_text        │  filter_text      │  text_edit        │  additional_text_edits  │
├───────────────────┼─────────────────┼─────────────────────┼───────────────────┼───────────────────┼───────────────────┼─────────────────────────┤
│  moduleNameQ      │   ...           │  Local              │  moduleNameQ      │  moduleNameQ      │  7:6..7:7         │  ...                    │
│                   │                 │                     │                   │                   │  moduleNameQ      │                         │
├───────────────────┼─────────────────┼─────────────────────┼───────────────────┼───────────────────┼───────────────────┼─────────────────────────┤
│  fullModuleNameQ  │   ...           │  Local              │  fullModuleNameQ  │  fullModuleNameQ  │  7:6..7:7         │  ...                    │
│                   │                 │                     │                   │                   │  fullModuleNameQ  │                         │
├───────────────────┼─────────────────┼─────────────────────┼───────────────────┼───────────────────┼───────────────────┼─────────────────────────┤
│  life             │   ...           │  Lib                │  life             │  life             │  7:6..7:7         │  ...                    │
│                   │                 │                     │                   │                   │  life             │                         │
├───────────────────┼─────────────────┼─────────────────────┼───────────────────┼───────────────────┼───────────────────┼─────────────────────────┤
│  life             │   (import Lib)  │  Lib                │  Lib              │  life             │  7:6..7:7         │  ...                    │
│                   │                 │                     │                   │                   │  life             │                         │
╰───────────────────┴─────────────────┴─────────────────────┴───────────────────┴───────────────────┴───────────────────┴─────────────────────────╯


Completion at Position { line: 10, character: 24 }

```
just :: forall a. a -> M
--                      ^
```

╭─────────┬──────────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬───────────────────────────────╮
│  label  │  label_detail        │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits        │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Maybe  │   ...                │  Lib                │  Maybe      │  Maybe        │  10:23..10:24  │  ...                          │
│         │                      │                     │             │               │  Maybe         │                               │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Mul    │   (import Prim.Int)  │  Prim.Int           │  Prim.Int   │  Mul          │  10:23..10:24  │  5:0..5:0                     │
│         │                      │                     │             │               │  Mul           │  import Prim.Int (class Mul)  │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Maybe  │   (import Lib)       │  Lib                │  Lib        │  Maybe        │  10:23..10:24  │  ...                          │
│         │                      │                     │             │               │  Maybe         │                               │
╰─────────┴──────────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴───────────────────────────────╯


Completion at Position { line: 12, character: 8 }

```
just = J
--      ^
```

╭─────────┬─────────────────┬─────────────────────┬─────────────┬───────────────┬──────────────┬─────────────────────────╮
│  label  │  label_detail   │  label_description  │  sort_text  │  filter_text  │  text_edit   │  additional_text_edits  │
├─────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼──────────────┼─────────────────────────┤
│  Just   │   ...           │  Lib                │  Just       │  Just         │  12:7..12:8  │  ...                    │
│         │                 │                     │             │               │  Just        │                         │
├─────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼──────────────┼─────────────────────────┤
│  Just   │   (import Lib)  │  Lib                │  Lib        │  Just         │  12:7..12:8  │  ...                    │
│         │                 │                     │             │               │  Just        │                         │
╰─────────┴─────────────────┴─────────────────────┴─────────────┴───────────────┴──────────────┴─────────────────────────╯


Completion at Position { line: 15, character: 22 }

```
nothing :: forall a. M
--                    ^
```

╭─────────┬──────────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬───────────────────────────────╮
│  label  │  label_detail        │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits        │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Maybe  │   ...                │  Lib                │  Maybe      │  Maybe        │  15:21..15:22  │  ...                          │
│         │                      │                     │             │               │  Maybe         │                               │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Mul    │   (import Prim.Int)  │  Prim.Int           │  Prim.Int   │  Mul          │  15:21..15:22  │  5:0..5:0                     │
│         │                      │                     │             │               │  Mul           │  import Prim.Int (class Mul)  │
├─────────┼──────────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼───────────────────────────────┤
│  Maybe  │   (import Lib)       │  Lib                │  Lib        │  Maybe        │  15:21..15:22  │  ...                          │
│         │                      │                     │             │               │  Maybe         │                               │
╰─────────┴──────────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴───────────────────────────────╯


Completion at Position { line: 17, character: 11 }

```
nothing = N
--         ^
```

╭───────────┬─────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label    │  label_detail   │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits  │
├───────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Nothing  │   ...           │  Lib                │  Nothing    │  Nothing      │  17:10..17:11  │  ...                    │
│           │                 │                     │             │               │  Nothing       │                         │
├───────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Nothing  │   (import Lib)  │  Lib                │  Lib        │  Nothing      │  17:10..17:11  │  ...                    │
│           │                 │                     │             │               │  Nothing       │                         │
╰───────────┴─────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴─────────────────────────╯


Completion at Position { line: 21, character: 12 }

```
refQ = Lib.l
--          ^
```

╭─────────┬────────────────┬─────────────────────┬─────────────┬───────────────┬───────────────┬─────────────────────────╮
│  label  │  label_detail  │  label_description  │  sort_text  │  filter_text  │  text_edit    │  additional_text_edits  │
├─────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼───────────────┼─────────────────────────┤
│  plus   │   ...          │  Lib                │  Lib.plus   │  Lib.plus     │  21:7..21:12  │  ...                    │
│         │                │                     │             │               │  Lib.plus     │                         │
├─────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼───────────────┼─────────────────────────┤
│  life   │   ...          │  Lib                │  Lib.life   │  Lib.life     │  21:7..21:12  │  ...                    │
│         │                │                     │             │               │  Lib.life     │                         │
╰─────────┴────────────────┴─────────────────────┴─────────────┴───────────────┴───────────────┴─────────────────────────╯


Completion at Position { line: 24, character: 29 }

```
justQ :: forall a. a -> Lib.M
--                           ^
```

╭─────────┬────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label  │  label_detail  │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits  │
├─────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Maybe  │   ...          │  Lib                │  Lib.Maybe  │  Lib.Maybe    │  24:24..24:29  │  ...                    │
│         │                │                     │             │               │  Lib.Maybe     │                         │
╰─────────┴────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴─────────────────────────╯


Completion at Position { line: 26, character: 8 }

```
justQ = J
--      ^
```

╭─────────┬─────────────────┬─────────────────────┬─────────────┬───────────────┬──────────────┬─────────────────────────╮
│  label  │  label_detail   │  label_description  │  sort_text  │  filter_text  │  text_edit   │  additional_text_edits  │
├─────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼──────────────┼─────────────────────────┤
│  Just   │   ...           │  Lib                │  Just       │  Just         │  26:8..26:9  │  ...                    │
│         │                 │                     │             │               │  Just        │                         │
├─────────┼─────────────────┼─────────────────────┼─────────────┼───────────────┼──────────────┼─────────────────────────┤
│  Just   │   (import Lib)  │  Lib                │  Lib        │  Just         │  26:8..26:9  │  ...                    │
│         │                 │                     │             │               │  Just        │                         │
╰─────────┴─────────────────┴─────────────────────┴─────────────┴───────────────┴──────────────┴─────────────────────────╯


Completion at Position { line: 29, character: 27 }

```
nothingQ :: forall a. Lib.M
--                         ^
```

╭─────────┬────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label  │  label_detail  │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits  │
├─────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Maybe  │   ...          │  Lib                │  Lib.Maybe  │  Lib.Maybe    │  29:22..29:27  │  ...                    │
│         │                │                     │             │               │  Lib.Maybe     │                         │
╰─────────┴────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴─────────────────────────╯


Completion at Position { line: 31, character: 16 }

```
nothingQ = Lib.N
--              ^
```

╭───────────┬────────────────┬─────────────────────┬───────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label    │  label_detail  │  label_description  │  sort_text    │  filter_text  │  text_edit     │  additional_text_edits  │
├───────────┼────────────────┼─────────────────────┼───────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Nothing  │   ...          │  Lib                │  Lib.Nothing  │  Lib.Nothing  │  31:11..31:16  │  ...                    │
│           │                │                     │               │               │  Lib.Nothing   │                         │
╰───────────┴────────────────┴─────────────────────┴───────────────┴───────────────┴────────────────┴─────────────────────────╯


Completion at Position { line: 34, character: 16 }

```
moduleNameQ = Li
--              ^
```

╭────────────┬────────────────┬─────────────────────┬─────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label     │  label_detail  │  label_description  │  sort_text  │  filter_text  │  text_edit     │  additional_text_edits  │
├────────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Lib       │   ...          │  Lib                │  Lib        │  Lib          │  34:14..34:16  │  ...                    │
│            │                │                     │             │               │  Lib           │                         │
├────────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  nothingQ  │   ...          │  Local              │  nothingQ   │  nothingQ     │  34:14..34:16  │  ...                    │
│            │                │                     │             │               │  nothingQ      │                         │
├────────────┼────────────────┼─────────────────────┼─────────────┼───────────────┼────────────────┼─────────────────────────┤
│  life      │   ...          │  Lib                │  life       │  life         │  34:14..34:16  │  ...                    │
│            │                │                     │             │               │  life          │                         │
╰────────────┴────────────────┴─────────────────────┴─────────────┴───────────────┴────────────────┴─────────────────────────╯


Completion at Position { line: 37, character: 24 }

```
fullModuleNameQ = Data.M
--                      ^
```

╭──────────────┬────────────────┬─────────────────────┬──────────────┬───────────────┬────────────────┬─────────────────────────╮
│  label       │  label_detail  │  label_description  │  sort_text   │  filter_text  │  text_edit     │  additional_text_edits  │
├──────────────┼────────────────┼─────────────────────┼──────────────┼───────────────┼────────────────┼─────────────────────────┤
│  Data.Maybe  │   ...          │  Data.Maybe         │  Data.Maybe  │  Data.Maybe   │  37:18..37:24  │  ...                    │
│              │                │                     │              │               │  Data.Maybe    │                         │
╰──────────────┴────────────────┴─────────────────────┴──────────────┴───────────────┴────────────────┴─────────────────────────╯
